packages/core/src/components/time-picker/time-picker.tsx

```diff
@@ -86,21 +129,25 @@ export class TimePicker {
 
   @Watch('time')
   watchTimePropHandler(newValue: string) {
-    this._time = DateTime.fromFormat(newValue, this.format);
-    if (!this._time.isValid) {
+    const timeFormat = DateTime.fromFormat(newValue, this.format);
+    if (!timeFormat.isValid) {
       throw new Error('Format is not supported or not correct');
     }
+
+    this._time = DateTime.fromFormat(newValue, this.format);
   }
 
   /**
    * Set time reference
+   *
+   * @deprecated This is determined by the currently set time
    */
-  @Prop() timeReference: 'AM' | 'PM' | undefined;
+  @Prop() timeReference: 'AM' | 'PM';
```

Comment:
Removing timeReference type `undefined` will break existing api description. It should still be optional and handled by the code.

```
@Prop() timeReference?: 'AM' | 'PM';
```